bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/handler/DaikinMadokaHandler.java

```diff
@@ -0,0 +1,761 @@
+
+        // Check that arguments are valid.
+        if (characteristic.getUuid() == null) {
+            return;
+        }
+
+        // We are only interested in the Notify Characteristic of UART service
+        if (!characteristic.getUuid().equals(DaikinMadokaBindingConstants.CHAR_NOTIF_UUID)) {
+            return;
+        }
+
+        // A message cannot be null or have a 0-byte length
+        if (characteristic.getByteValue() == null || characteristic.getByteValue().length == 0) {
+            return;
+        }
+
+        this.uartProcessor.chunkReceived(characteristic.getByteValue());
+
+    }
+
+    private void submitCommand(BRC1HCommand command) {
+        Executor executor = commandExecutor;
+
+        if (executor != null) {
+            executor.execute(() -> processCommand(command));
+        }
+    }
+
+    private void processCommand(BRC1HCommand command) {
+        logger.debug("[{}] ProcessCommand {}", super.thing.getUID().getId(), command.getClass().getSimpleName());
+
+        try {
+            currentCommand = command;
+            uartProcessor.abandon();
+
+            if (command.getRequest() == null) {
+                logger.debug("Unable to send command {} to device {}: Request NULL.",
+                        command.getClass().getSimpleName(), device.getAddress());
+                command.setState(BRC1HCommand.State.FAILED);
+                return;
+            }
+
+            if (device == null || device.getConnectionState() != ConnectionState.CONNECTED) {
+                logger.debug("Unable to send command {} to device {}: not connected",
+                        command.getClass().getSimpleName(), address);
+                command.setState(BRC1HCommand.State.FAILED);
+                return;
+            }
+
+            if (!resolved) {
+                logger.debug("Unable to send command {} to device {}: services not resolved",
+                        command.getClass().getSimpleName(), device.getAddress());
+                command.setState(BRC1HCommand.State.FAILED);
+                return;
+            }
+
+            BluetoothCharacteristic charWrite = device
+                    .getCharacteristic(DaikinMadokaBindingConstants.CHAR_WRITE_WITHOUT_RESPONSE_UUID);
+            if (charWrite == null) {
+                logger.warn("Unable to execute {}. Characteristic '{}' could not be found.",
+                        command.getClass().getSimpleName(),
+                        DaikinMadokaBindingConstants.CHAR_WRITE_WITHOUT_RESPONSE_UUID);
+                command.setState(BRC1HCommand.State.FAILED);
+                return;
+            }
+
+            BluetoothCharacteristic charNotif = this.device
+                    .getCharacteristic(DaikinMadokaBindingConstants.CHAR_NOTIF_UUID);
+
+            if (charNotif != null) {
+                @NonNull
+                BluetoothCharacteristic c = charNotif;
+                device.enableNotifications(c);
```

Comment:
What is wrong with doing this?
```suggestion
                device.enableNotifications(charNotif);
```

Reply:
No idea! I might have been drunk when coding this :-D